src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,19 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initcialState = {
+  categories: [],
+  state: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState: initcialState,
+  reducers: {
+    stateCheck: (state) => {
+      state.state = 'Under construction';
+    },
+  },
+});
```

Comment:
- [ ] Ouch! this is close but you seem to be missing something in the `stateCheck` reducer. As part of the project requirements, the reducer should check the status first before updating the state so to fix this, consider accepting an action as a parameter in the `stateCheck` reducer and checking the payload of that action to update the state. This will ensure that your reducer behaves correctly and follows the principles of Redux. Keep up the good work!
- The reducer should looks something like this:
```jsx
stateCheck: (state, action) => {
      state.state =
        action.payload === "Under construction"
          ? "Under construction"
          : state.state
    },
```
- Here's the project requirement as a reminder:
  ![image](https://github.com/Mofuhidy/bookstore-react/assets/104968236/a422ee84-7fb4-4a5e-a9b3-5d7ed9839660)